cmake/AddFdbTest.cmake

```diff
@@ -375,3 +375,92 @@ function(package_bindingtester)
   add_custom_target(bindingtester ALL DEPENDS ${tar_file})
   add_dependencies(bindingtester copy_bindingtester_binaries)
 endfunction()
+
+function(add_fdbclient_test)
+  set(options DISABLED ENABLED)
+  set(oneValueArgs NAME)
+  set(multiValueArgs COMMAND)
+  cmake_parse_arguments(T "${options}" "${oneValueArgs}" "${multiValueArgs}" "${ARGN}")
+  if(OPEN_FOR_IDE)
+    return()
+  endif()
+  if(NOT T_ENABLED AND T_DISABLED)
+    return()
+  endif()
+  if(NOT T_NAME)
+    message(FATAL_ERROR "NAME is a required argument for add_fdbclient_test")
+  endif()
+  if(NOT T_COMMAND)
+    message(FATAL_ERROR "COMMAND is a required argument for add_fdbclient_test")
+  endif()
+  message(STATUS "Adding Client test ${T_NAME}")
+  add_test(NAME "${T_NAME}"
+    COMMAND ${CMAKE_SOURCE_DIR}/tests/TestRunner/tmp_cluster.py
+            --build-dir ${CMAKE_BINARY_DIR}
+            --
+            ${T_COMMAND})
+  set_tests_properties("${T_NAME}" PROPERTIES TIMEOUT 60) 
+endfunction()
+
+function(add_multi_fdbclient_test)
+  # Will Create 3 separate clusters before running the remaining command
+  set(options DISABLED ENABLED)
+  set(oneValueArgs NAME)
+  set(multiValueArgs COMMAND)
+  cmake_parse_arguments(T "${options}" "${oneValueArgs}" "${multiValueArgs}" "${ARGN}")
+  if(OPEN_FOR_IDE)
+    return()
+  endif()
+  if(NOT T_ENABLED AND T_DISABLED)
+    return()
+  endif()
+  if(NOT T_NAME)
+    message(FATAL_ERROR "NAME is a required argument for add_fdbclient_test")
+  endif()
+  if(NOT T_COMMAND)
+    message(FATAL_ERROR "COMMAND is a required argument for add_fdbclient_test")
+  endif()
+  message(STATUS "Adding Client test ${T_NAME}")
+  add_test(NAME "${T_NAME}"
+    COMMAND ${CMAKE_SOURCE_DIR}/tests/TestRunner/tmp_multi_cluster.py
```

Comment:
This script is probably forgetting to call `kill 0` at exit.

This SO thread seems to suggest python has support for this, but doesn't handle re-entrancy on its own: https://stackoverflow.com/questions/28176215/execute-code-at-exit

Here's some bash code that does the right thing, but it won't run on windows:
```
#!/bin/bash

trap "kill 0" EXIT

spawn_subprocess_with_immunity &

run_something_that_must_be_killed_if_we_are_killed
```
On the other hand, `kill 0` also won't work on windows...

Reply:
my understanding (limited though it probably is) is that `ctest` essentially does a `kill -9` on the subprocess if there's a  timeout; there is no guarantee that our underlying scripts will be able to recover from that type of failure.